clients/githubrepo/checkruns_e2e_test.go

```diff
@@ -48,4 +50,39 @@ var _ = Describe("E2E TEST: githubrepo.checkrunsHandler", func() {
 			Expect(*checkrunshandler.checkData.RateLimit.Cost).Should(BeNumerically("<=", 1))
 		})
 	})
+	Context("E2E TEST: listCheckRunsForRef", func() {
+		It("Should return check runs for a valid ref", func() {
+			repourl := &repoURL{
+				owner:     "ossf",
+				repo:      "scorecard",
+				commitSHA: clients.HeadSHA,
+			}
+			httpClient := &http.Client{}
+			client := github.NewClient(httpClient)
```

Comment:
Without passing any token, I would expect the quota/limits to be based on the IP address, so this may have intermittent authentication issues